components/editor/utils/keyHandlers.js

```diff
@@ -0,0 +1,155 @@
+import { Block } from 'slate'
+import { matchBlock } from './'
+import { getClosestInSelection } from './selection'
+
+const focusNext = change => {
+  const { value } = change
+  const nextBlock = value.document.getNextBlock(value.endBlock.key)
+  if (nextBlock) {
+    return change.collapseToStartOf(
+      nextBlock
+    )
+  }
+  return true
+}
+
+const focusPrevious = change => {
+  const { value } = change
+  const nextBlock = value.document.getPreviousBlock(value.endBlock.key)
+  if (nextBlock) {
+    return change.collapseToEndOf(
+      nextBlock
+    )
+  }
+  return true
+}
+
+const insertAfter = (change, afterType, insertAfterType) => {
+  const { value } = change
+  const rootNode = (
+      insertAfterType &&
+      value.document.getFurthest(
+        value.endBlock.key, matchBlock(insertAfterType)
+      )
+    ) ||
+    value.document
+
+  const index = rootNode.nodes.findIndex(
+    n => !!n.findDescendant(m => m.key === value.endBlock.key)
+  )
+
+  if (index !== -1) {
+    return change.insertNodeByKey(
+      rootNode.key,
+      index + 1,
+      Block.create({ type: afterType })
+    )
+  }
+  return change
+}
+
+export const staticKeyHandler = ({ TYPE, rule }) => {
+  const { afterType, insertAfterType } = rule.editorOptions
+
+  return (event, change) => {
+    const isBackspace = event.key === 'Backspace'
+    const isEnter = event.key === 'Enter'
+    const isTab = event.key === 'Tab'
+    const isDelete = event.key === 'Delete'
+
+    if (!isBackspace && !isEnter && !isDelete && !isTab) return
+    const { value } = change
+    const inSelection = value.blocks.some(matchBlock(TYPE))
+
+    if (inSelection && value.isExpanded) {
```

Comment:
this feels to invasive: static one can now only be delete on character at a time. Could we support deleting a range if it is exclusively within one/our block? And when not first reduce the range to just that block? so double backspace would always work to clear a complete lead / credit line